sa/model.go

```diff
@@ -88,14 +88,18 @@ func ClearEmail(ctx context.Context, dbMap db.DatabaseMap, regID int64, email st
 			return nil, nil
 		}
 
-		currPb.Contact = newContacts
-		newModel, err := registrationPbToModel(currPb)
-		if err != nil {
-			return nil, err
-		}
+		// UPDATE the row with a direct database query, in order to avoid LockCol issues.
+		//
+		// TODO(#7716): Revert to use tx.Update() once LockCol has been removed.
+		_, err = dbMap.ExecContext(ctx,
+			"UPDATE registrations SET contact = ? WHERE id = ? LIMIT 1",
+			newContacts,
+			regID,
+		)
 
-		return tx.Update(ctx, newModel)
+		return nil, err
```

Comment:
We may also want to improve the error checking here. While simply returning `nil, err` here is certainly semantically correct, it's somewhat surprising to the reader -- they have to think through "wait, are we _always_ returning an error here? what if err is nil?". One way to make this both more idiomatic _and_ to catch even more cases would be to use the pattern in `sa.SetOrderError`:

```go
res, err := tx.ExecContext(...)
if err != nil {
  return nil, err
}
rowsAffected, err := res.RowsAffected()
if err != nil || rowsAffected != 1 {
  return nil, some error here
}
return nil, nil